collect_app/src/test/java/org/odk/collect/android/preferences/JsonPreferencesGeneratorTest.kt

```diff
@@ -142,11 +153,11 @@ class JsonPreferencesGeneratorTest {
         settingsProvider.getAdminSettings("2").saveAll(adminPrefsForProjectX)
 
         // Verify the demo project
-        verifyJsonContent(jsonPreferencesGenerator.getJSONFromPreferences(), generalPrefsForDemoProject, adminPrefsForDemoProject)
+        verifyJsonContent(jsonPreferencesGenerator.getJSONFromPreferences(), generalPrefsForDemoProject, adminPrefsForDemoProject, projectDetails)
```

Comment:
This all really confused me as the test is using a real `SettingsProvider` which means it uses the real `CurrentProjectProvider` but then the actual object under test uses a mock `CurrentProjectProvider`. I think we need to clean that up as the boundaries of this test aren't quite defined.

One idea is that we create a test settings provider implementation (and pull out an interface to sit above the real and test one) that just uses `InMemSettings` under the hood and doesn't use the `CurrentProjectProvider` (it just returns different settings objects for different project Ids and a default one for no project Id). This would be really useful in a lot of places as we could use it instead of the real `SettingsProvider` and potentially avoid Robolectric as well (as we need it for `SharedPreferences`).